Firestore/core/src/firebase/firestore/model/field_path.h

```diff
@@ -47,10 +48,28 @@ class FieldPath : public impl::BasePath<FieldPath> {
   FieldPath(const IterT begin, const IterT end) : BasePath{begin, end} {
   }
   FieldPath(std::initializer_list<std::string> list) : BasePath{list} {
+    if (empty()) {
+      api::ThrowInvalidArgument(
+          "Invalid field path. Provided names must not be empty.");
+    }
+
+    for (size_t i = 0; i < size(); i++) {
```

Comment:
Optional nit (and sorry I missed it on the first pass): this is basically [`std::any_of`](https://en.cppreference.com/w/cpp/algorithm/all_any_none_of):
```cpp
if (std::any_of(begin(), end(), [](const std::string& segment) { return segment.empty(); }) {
  // Throw
}
```

Reply:
Thanks, though since we use the index in the error message this would result in a slightly-less useful error.  So I'm inclined to leave it as-is.

Reply:
Also, `std::any_of` doesn't save too much typing of the for loop, and the for loop is straightforwardly portable to the other platforms.